packages/http-message-sig/src/types.ts

```diff
@@ -66,9 +72,15 @@ export type Component =
   | "@query"
   | "@query-param"
   | "@status"
-  | "@request-response"
   | string;
 
+export type ComponentWithParameters = {
+  name: Component;
+  parameters?: ComponentParameters;
+}
```

Comment:
is a raw type the right interface? I feel that this adds quite some complexity, as seen in the `.map(name => ({name}))` below for instance.